File: app/src/main/java/com/imangazalievm/bubbble/data/repository/FollowersRepository.java
package com.imangazalievm.bubbble.data.repository;

import com.imangazalievm.bubbble.data.network.DribbbleApiService;
import com.imangazalievm.bubbble.domain.models.Follow;
import com.imangazalievm.bubbble.domain.models.UserFollowersRequestParams;
import com.imangazalievm.bubbble.domain.repository.IFollowersRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class FollowersRepository implements IFollowersRepository {

    private DribbbleApiService dribbbleApiService;

    @Inject
    public FollowersRepository(DribbbleApiService dribbbleApiService) {
        this.dribbbleApiService = dribbbleApiService;
    }

    public Single<List<Follow>> getUserFollowers(UserFollowersRequestParams requestParams) {
        return dribbbleApiService.getUserFollowers(requestParams.getUserId(),
                requestParams.getPage(), requestParams.getPageSize());
    }

}

File: app/src/main/java/com/bubbble/data/users/FollowersRepository.kt
package com.bubbble.data.users

import com.bubbble.core.models.user.Follow
import com.bubbble.core.models.user.UserFollowersParams
import com.bubbble.core.network.DribbbleApi
import javax.inject.Inject

class FollowersRepository @Inject constructor(
    private val dribbbleApi: DribbbleApi
) {

    suspend fun getUserFollowers(params: UserFollowersParams): List<Follow> {
        return dribbbleApi.getUserFollowers(
            params.userId,
            params.page,
            params.pageSize
        )
    }

}